number_parser/parser.py

```diff
@@ -226,6 +246,8 @@ def parse(input_string, language='en'):
 
     for token in tokens:
         compare_token = _strip_accents(token.lower())
+        ordinal_number = _is_ordinal_token(compare_token, lang_data)
```

Comment:
This parameter is added and used to break the processing when we encounter an ordinal number as it indicates the end of current processing. This is to ensure cases like ```twentieth seventh fiftieth third ``` work

Reply:
I can add a similar sort of logic to ```parse_ordinal``` to break on encountering an ordinal number. This would then reject mixed cardinal/ordinal like  ``` twentieth three ```